libs/libc/pthread/pthread_rwlock_wrlock.c

```diff
@@ -113,6 +113,12 @@ int pthread_rwlock_clockwrlock(FAR pthread_rwlock_t *rw_lock,
         {
           err = pthread_cond_clockwait(&rw_lock->cv, &rw_lock->lock,
                                        clockid, ts);
+
+          if (err == EINTR)
+            {
+              err = 0;
```

Comment:
why need zero err here?

Reply:
might exit loop after line 110, so should clear err.

Reply:
in this case, we should return EINTR, not error.